common/src/main/java/feast/common/logging/AuditLogger.java

```diff
@@ -0,0 +1,149 @@
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+import org.slf4j.event.Level;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class AuditLogger {
+  private static final Marker AUDIT_MARKER = MarkerFactory.getMarker("AUDIT_MARKER");
+  private static FeastInstance instance;
+  private static AuditLogProperties properties;
+
+  @Autowired
+  public AuditLogger(FeastInstance instance, LoggingProperties properties) {
+    // Spring runs this constructor when creating the AuditLogger bean,
+    // which allows us to populate the AuditLogger class with dependencies.
+    // This allows us to use the dependencies in the AuditLogger's static methods
+    AuditLogger.instance = instance;
+    AuditLogger.properties = properties.getAudit();
+  }
+
+  /**
+   * Log the handling of a Protobuf message by a service call.
+   *
+   * @param level describing the severity of the log.
+   * @param method the name of the method handling the service call
+   * @param request protobuf message in the service call.
+   * @param response protobuf message in the servicef call.
+   */
+  public static void logMessage(
+      Level level, String method, Message request, Message response, String identity) {
+    log(
+        level,
+        MessageAuditLogEntry.newBuilder()
+            .setInstance(instance)
+            .setMethod(method)
+            .setRequest(request)
+            .setResponse(response)
+            .setIdentity(identity)
+            .build());
+  }
+
+  /**
+   * Log an action being taken on a specific resource
+   *
+   * @param level describing the severity of the log.
+   * @param action name of the action being taken on specific resource.
+   * @param resourceType the type of resource being logged.
+   * @param resourceId resource specific identifier identifing the instance of the resource.
+   */
+  public static void logAction(
+      Level level, String action, ResourceType resourceType, String resourceId) {
+    log(
+        level,
+        ActionAuditLogEntry.newBuilder()
+            .setInstance(instance)
+            .setAction(action)
+            .setResource(LogResource.of(resourceType, resourceId))
+            .build());
+  }
+
+  /**
+   * Log a transition in state/status in a specific resource.
+   *
+   * @param level describing the severity of the log.
+   * @param status name of end status which the resource transition to.
+   * @param resourceType the type of resource being logged.
+   * @param resourceId resource specific identifier identifing the instance of the resource.
+   */
+  public static void logTransition(
+      Level level, String status, ResourceType resourceType, String resourceId) {
+    log(
+        level,
+        TransitionAuditLogEntry.newBuilder()
+            .setInstance(instance)
+            .setStatus(status)
+            .setResource(LogResource.of(resourceType, resourceId))
```

Comment:
Isn't it cleaner to have a ResourceType and a String resourceId, and not a `LogResource`? Seems like an unnecessary abstraction.

Reply:
The idea was that we consolidate referencing resources into this format across AuditLogEntry subclasses. Both `Action` and `Transition` `AuditLogEntry`s use LogResource to reference a resource.

Reply:
`LogResource` consolidates the resource referencing fields together. It also looks cleaner on the JSON ie 
`entry.resource.id` and `entry.resource.type` instead of `entry.resourceId` and `entry.resourceType`